### PR TITLE
Default resource namespace when it is missing

### LINKED ISSUE #19: ManagedResource cannot apply when resource is without explicit namespace
Since 0.6.0 the namespace for the ManagedResource resources is required. If namespace is not provided, the g/gardener-resource-manager fails to apply the resources.

/kind bug

Steps to reproduce:
1. Apply ManagedResource with resource without namespace
```bash
$ cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Secret
metadata:
  name: managedresource-example1
type: Opaque
stringData:
  objects.yaml: |
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: test-1234
      annotations:
        resources.gardener.cloud/ignore: "true"
    ---
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: test-5678
EOF


$ cat <<EOF | kubectl apply -f -
apiVersion: resources.gardener.cloud/v1alpha1
kind: ManagedResource
metadata:
  name: example
  namespace: default
spec:
  secretRefs:
  - name: managedresource-example1
EOF
```

2. Ensure that the apply fails.

```bash
$ make start
{"level":"info","ts":"2019-11-07T16:46:00.297+0200","logger":"gardener-resource-manager.reconciler","msg":"Applying","resource":"v1/ConfigMap/default/test-1234"}
{"level":"error","ts":"2019-11-07T16:46:00.312+0200","logger":"controller-runtime.controller","msg":"Reconciler error","controller":"resource-controller","request":"default/example","error":"Errors occurred during applying: [error during apply of object \"v1/ConfigMap/default/test-1234\": the server does not allow this method on the requested resource error during apply of object \"v1/ConfigMap/default/test-5678\": the server does not allow this method on the requested resource]",
```

/cc @vpnachev

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
Default resource namespace when it is missing.

**Which issue(s) this PR fixes**:
Fixes #19

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/controller/managedresources/controller.go [hunk 1]
```diff
@@ -151,9 +151,14 @@ func (r *Reconciler) reconcile(mr *resourcesv1alpha1.ManagedResource, log logr.L
 					continue
 				}
 
+				obj := &unstructured.Unstructured{Object: decodedObj}
+				if obj.GetKind() != "Namespace" && obj.GetNamespace() == "" {
```
Comment (vpnachev) [inline pkg/controller/managedresources/controller.go]: `obj.GetKind() != "Namespace"` hmm, what about for cluster level resources like `ClusterRole` ?

Comment (ialidzhikov) [reply pkg/controller/managedresources/controller.go]: Non-namespaced resources are applied but in the status are shown with ns as:
```
    - apiVersion: rbac.authorization.k8s.io/v1
      kind: ClusterRole
      name: role-grantor
      namespace: default
```

Probably @rfranzke could share what was the motivation removing the namespace defaulting in 77126f1274ae7dfb524b0c6a2a47f96514cf7e07.

Comment (vpnachev) [reply pkg/controller/managedresources/controller.go]: So it will be set anyway, simply the API will ignore it. @mandelsoft  @rfranzke, am I right?

**Comment by rfranzke** [comment]
/hold
I have to check again why we have done this - I remember we had some problems with defaulting.

**Comment by rfranzke** [comment]
I think the main problem back then was that the object key generation function did not default the namespace. Hence, should be good to go.

**Pull request merged**